### PR TITLE
friends-app

### PR DESCRIPTION
**Pull request by AntonKottans**
friends-app

[Demo](https://antonkottans.github.io/friends-app/) |
[Code base](https://github.com/AntonKottans/friends-app)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

**Comment by suchov** [comment]
Nice, I like the refactoring one - it got more readable and looks nicer now!

File: submissions/antonkottans/friends-app/script.js [hunk 1]
```diff
@@ -0,0 +1,595 @@
...
+    let rows = Math.trunc(containerHeight / cardSize.height) || 1;
+    if (containerHeight % cardSize.height <= gridGap * (rows - 1)) rows--;
+    let amountPerPage = columns * rows;
+    let amountOfPages = Math.ceil(amountOfElements / amountPerPage);
+
+    return { amountPerPage, amountOfPages };
+};
+
+const getNearestIntegers = (
+    currentNumber = 1,
```
Comment (zonzujiro) [inline submissions/antonkottans/friends-app/script.js:72]:
```suggestion
    currentPage = 1,
```

File: submissions/antonkottans/friends-app/script.js [hunk 2]
```diff
@@ -0,0 +1,595 @@
...
+    amountOfNumbersToReturn_odd = 1,
+    totalNumbers = 1
+) => {
+    const centralNumber = parseInt(currentNumber);
+    let amountOfNumbers = parseInt(amountOfNumbersToReturn_odd);
+    if (!(amountOfNumbersToReturn_odd % 2)) amountOfNumbers -= 1;
+    let numbersArr = [currentNumber];
+    if (
+        centralNumber === centralNumber &&
+        amountOfNumbers === amountOfNumbers &&
```
Comment (zonzujiro) [inline submissions/antonkottans/friends-app/script.js:82]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN

**Review by zonzujiro — CHANGES_REQUESTED** [review]

File: submissions/antonkottans/friends-app/index.html [hunk 3]
```diff
@@ -0,0 +1,94 @@
+<!DOCTYPE html>
+<html lang="en">
+    <head>
+        <meta charset="UTF-8" />
+        <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+        <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+        <link rel="stylesheet" href="./css/main.css" />
+        <title>Document</title>
```
Comment (A-Ostrovnyy) [inline submissions/antonkottans/friends-app/index.html:8]: Add propper page title.

File: submissions/antonkottans/friends-app/index.html [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+                <button class="sort-menu-button" id="name-ascending"></button>
+
+                <button class="sort-menu-button" id="name-descending"></button>
+            </div>
+            <div class="filter-menu">
+                <div class="age-filter-container">
+                    <input
+                        class="age-input"
+                        type="number"
+                        name=""
```
Comment (A-Ostrovnyy) [inline submissions/antonkottans/friends-app/index.html:33]: An empty value for the input's `name` attribute is a bad practice.

File: submissions/antonkottans/friends-app/index.html [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+                        <label class="sex-filter-label" for="female">
+                            Female
+                        </label>
+                        <input
+                            type="radio"
+                            class="radio-sex-input"
+                            name="radiobutton-sex"
+                            id="all"
+                            checked
+                        />
```
Comment (A-Ostrovnyy) [inline submissions/antonkottans/friends-app/index.html:80]: Add `value` attribute with the corresponding value.

File: submissions/antonkottans/friends-app/sass/components.scss [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+    overflow: hidden;
+
+    .loading-container{
+        .loading-text{
+            color: white;
+            font-size: 30px;
+        }
+        .loading-dot{
+            background-color: $menu-background;
+            &.in-progress{
```
Comment (A-Ostrovnyy) [inline submissions/antonkottans/friends-app/sass/components.scss:154]: Avoid too specific CSS selectors. [Read more](https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity) about CSS spesifity

File: submissions/antonkottans/friends-app/sass/components.scss [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+$body-background: #014546;
+$dark-gradient-color: #001d1e;
+$light-gradient-color: #014546;
+$light-gradient-color-hover: #236768;
+$text-color: #cccccc;
+$inputs-background-color: rgba(200, 200, 200, 0.5);
+$radio-pagination-color: #62ba85;
+$radio-pagination-color-checked: #a6fec9;
+$radio-pagination-background-color-checked: #003435;
+$radio-pagination-color-hover: #84dca7;
```
Comment (A-Ostrovnyy) [inline submissions/antonkottans/friends-app/sass/components.scss:15]: :fire:

File: submissions/antonkottans/friends-app/script.js [hunk 8]
```diff
@@ -0,0 +1,595 @@
...
+        })
+    ).then(() => {
+        state.loading = false;
+        updateContentAccordingToActiveFilters();
+    });
+};
+
+let createFriendCard = (friend) => {
+    const createFriendContainer = () => {
+        const friendCard = document.createElement("div");
```
Comment (A-Ostrovnyy) [inline submissions/antonkottans/friends-app/script.js:369]: For items like this better to use tag `article`

File: submissions/antonkottans/friends-app/script.js [hunk 9]
```diff
@@ -0,0 +1,595 @@
...
+        friendCard.classList.add("friend-container", "hidden");
+
+        return friendCard;
+    };
+
+    const addName = (friendCard) => {
+        const friendName = document.createElement("span");
```
Comment (A-Ostrovnyy) [inline submissions/antonkottans/friends-app/script.js:376]: If you will use `article` tag you should use the heading (h2 for example) for card title.

File: submissions/antonkottans/friends-app/script.js [hunk 10]
```diff
@@ -0,0 +1,595 @@
...
+
+    const addDOB = (friendCard) => {
+        const friendDOB = document.createElement("span");
+        friendDOB.textContent = `I am ${friend.dob.age} years old.`;
+        friendCard.appendChild(friendDOB);
+        return friendCard;
+    };
+
+    const addEmail = (friendCard) => {
+        const friendEmail = document.createElement("span");
```
Comment (A-Ostrovnyy) [inline submissions/antonkottans/friends-app/script.js:400]: For email better to use a link with a special value for `href` attribute. [Read more](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#linking_to_an_email_address)

File: submissions/antonkottans/friends-app/script.js [hunk 11]
```diff
@@ -0,0 +1,595 @@
...
+        friendEmail.textContent = friend.email;
+        friendEmail.classList.add("email");
+        friendCard.appendChild(friendEmail);
+        return friendCard;
+    };
+
+    const addCell = (friendCard) => {
+        const friendCell = document.createElement("span");
```
Comment (A-Ostrovnyy) [inline submissions/antonkottans/friends-app/script.js:408]: For phone numbers better to use a link with a special value for `href` attribute. [Read more](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#linking_to_telephone_numbers)

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
Good work!

File: submissions/antonkottans/friends-app/script.js [hunk 12]
```diff
@@ -0,0 +1,582 @@
...
+  let columns = Math.trunc(containerWidth / cardSize.width) || 1;
+  if (containerWidth % cardSize.width <= gridGap * (columns - 1)) columns--;
+  let rows = Math.trunc(containerHeight / cardSize.height) || 1;
+  if (containerHeight % cardSize.height <= gridGap * (rows - 1)) rows--;
+  let amountPerPage = columns * rows;
+  let amountOfPages = Math.ceil(amountOfElements / amountPerPage);
+  return { amountPerPage, amountOfPages };
+};
+
+const getNearestNaturalNumbers = (
```
Comment (A-Ostrovnyy) [inline submissions/antonkottans/friends-app/script.js:65]: Do you have float numbers in your work?)

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
Good progress

File: submissions/antonkottans/friends-app/script.js (hunk 12 above)
Comment (AntonKottans) [reply submissions/antonkottans/friends-app/script.js:65]: no, but this function doesn`t return zero or negative numbers.

**Review by A-Ostrovnyy — APPROVED** [review]
Well done!

**Pull request merged**